src/ViewModel/TextArea.php

```diff
@@ -45,10 +48,21 @@ public function __construct(
         int $rows = null,
         string $form = null,
         string $state = null,
-        string $message = null
+        string $message = null,
+        string $messageId = null
     ) {
         Assertion::nullOrChoice($state, [self::STATE_ERROR, self::STATE_VALID]);
 
+        if ($state === self::STATE_VALID && $messageId) {
+            throw new InvalidArgumentException('There must not be a messageId if the state is valid.');
```

Comment:
Maybe this shouldn't be the case? We might want to provide help info alongside the field. Though an error message would override it...

Reply:
Will update corresponding logic in elifesciences/pattern-library#752

Reply:
@thewilkybarkid Have updated the pattern library to allow info messages (e.g. https://github.com/elifesciences/pattern-library/pull/752/files#diff-4192423a99bed31822a1c2cbec6fd2e2R65). Currently, this would require an additional state: `const STATE_VALID = 'info';` in the relevant view models. Not sure that's quite right though, as it's the message that's informational, not the form item the message belongs to.  Checking before I make the required changes here, are you happy to go down this route?

Reply:
Don't think `info`'s needed, state is either valid, invalid, or not-yet-completed (currently no state).